### ISSUE TITLE
Suppress irrelevant warnings and messages in tests

### ISSUE THREAD

**Issue by IndrajeetPatil** [easystats/easystats#338]
## Preamble

Currently, running tests locally or on CI produces a number of warnings or messages, which are irrelevant for our testing purposes. That is, these are not warnings/messages that our functions produce, but rather produced by whatever modeling package we are using for testing. 

This makes it difficult to sort the wheat from the chaff when tests fail because one needs to trawl through a number of warnings and messages to get to the actual failures. 

We should instead either be using test examples that don't throw exceptions, or silence them while running tests.

## Examples

Here are a few randomly selected examples of warnings from across the packages in the ecosystem:

```r
Package version inconsistency detected.
TMB was built with Matrix version 1.4.1
Current Matrix version is 1.5.1
Please re-install 'TMB' from source using install.packages('TMB', type = 'source') or ask CRAN for a binary version of 'TMB' matching CRAN's 'Matrix' package
Backtrace:
  1. base::require("glmmTMB")
       at test-estimate_predicted.R:10:0
  6. base::library(...)
 11. base::loadNamespace(package, lib.loc)
 14. base::loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]])
 15. base (local) runHook(".onLoad", env, package.lib, package)
 20. TMB (local) fun(libname, pkgname)
 21. TMB:::checkMatrixPackageVersion()
```

```r
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
```

```r
glm.fit: fitted rates numerically 0 occurred
```

```r
glm.fit: algorithm did not converge
```

## Progress tracker

- [x] insight 
- [x] modelbased
- [x] bayestestR
- [x] effectsize
- [x] parameters
- [x] performance
- [x] correlation
- [x] report
- [x] see
- [x] datawizard
- [x] easystats

**Comment by IndrajeetPatil** [comment]
As usual, `{effectsize}` is perfect 🥹

**Comment by IndrajeetPatil** [comment]
I have created a new GHA [workflow](https://github.com/easystats/workflows/blob/main/.github/workflows/check-test-warnings.yaml) to catch warnings, so, to fix this issue for your repo, you just need to make sure that this workflow has green status. 

I can't think of a way to catch messages; maybe this is something each maintainer can check locally and take action accordingly. But getting rid of message is definitely low priority.

**Comment by IndrajeetPatil** [comment]
This workflow seems to fail if you are running tests in parallel. I am not sure why.

So, unless you fix the warnings, you can remove the following from `DESCRIPTION`:

```
Config/testthat/parallel: true
```

And once warnings have been taken care of, just delete the workflow and run tests in parallel again. 

Sorry for the trouble. Let me know if you can figure out why setting `options(warn = 2)` might create issues for running tests in parallel.

**Comment by IndrajeetPatil** [comment]
And if you think Indra is unnecessarily creating extra work for us, I do the same for each of my personal repos ([e.g.](https://github.com/IndrajeetPatil/statsExpressions/tree/main/.github/workflows)) as well and work to make sure that all workflows are running successfully. 😭

**Comment by strengejacke** [comment]
I think even more annoying is all the junk output from snapshot tests...
https://github.com/easystats/parameters/actions/runs/3931656465/jobs/6723197896

**Comment by IndrajeetPatil** [comment]
```r
── Warning ('test-glmmTMB.R:476'): print-model_parameters glmmTMB CI alignment ──
Adding new snapshot:
```

This means you actually haven't included a snapshot against which the output can be compared, so every time the test is run, it creates a new snapshot.

There is no snapshot corresponding to this test here:
https://github.com/easystats/parameters/tree/main/tests/testthat/_snaps

---

100% of tests I use in _statsExpressions_ and _ggstatsplot_ are snapshot tests, and I see [0 messages or warnings](https://github.com/IndrajeetPatil/statsExpressions/actions/runs/3920153244/jobs/6701601693). If you use them correctly, you won't see any "junk".

Also, I am getting tired of all this criticism of snapshot testing. If you don't like them, please don't use them in the repos you maintain. It's as simple as that.

**Comment by strengejacke** [comment]
But it used to work, not sure why they're now suddenly failing?

**Comment by IndrajeetPatil** [comment]
All snapshot test failures have now disappeared.

Most of the remaining failures have to do with `mmrm` models:
https://github.com/easystats/parameters/actions/runs/3933266126/jobs/6726788438

There are a couple of errors related to `get_data()`:

> Error in ``[.data.frame`(get_data(x, verbose = FALSE), find_random(x, split_nested = TRUE, 
    flatten = TRUE))`: undefined columns selected

**Comment by strengejacke** [comment]
> All snapshot test failures have now disappeared.

Thanks a lot! :-)
<sub>(there will be the day where I make peace with snapshot tests, I promise)</sub>

> There are a couple of errors related to get_data():

Yeah, I already tested against the changes in `get_data()`, so I'm not sure where these come from. Maybe from `n_grouplevels()`, I'll heck.

**Comment by rempsyc** [comment]
`report` should be good now

**Comment by IndrajeetPatil** [comment]
That's what we are after 😻 

No messages, no warnings, no errors.

<img width="639" alt="Screenshot 2023-03-23 at 08 19 28" src="https://user-images.githubusercontent.com/11330453/227131270-434891f0-e677-4973-9b0b-79b199514ea1.png">

**Comment by etiennebacher** [comment]
Looks like `modelbased` has clean tests already, there are some failures but they are related to examples in older Ubuntu. Similarly for `bayestestR` there are some errors/warnings in tests but there shouldn't be irrelevant messages

**Comment by IndrajeetPatil** [comment]
[modelbased](https://github.com/easystats/modelbased/actions/runs/4494901712/jobs/7907961674) is still spewing a ton of messages and warnings.

**Comment by etiennebacher** [comment]
Indeed, I didn't notice because this happens only in the random order testing
